### DISCUSSION TITLE
How to work with multiple environments in the same branch for image automation?

### DISCUSSION THREAD

**Discussion by ilya-git** [fluxcd/flux2#1099] [Q&A] [answered]
I have a gitops repository with the following folder structure:

- my repository
     - dev1
     - dev2
     - prod

I have 3 k8s clusters each with a GitRepository pointing to the same repository and having different "path". Now when I do the same for `ImageUpdateAutomation` it does not actually have a path, meaning each of the 3 clusters will perform parallel image update automation. I.e. `dev1` and `dev2` can point to the same image for one of the deployments. As soon as new image is available, 2 clusters will try to update code with the new tag.

Is it kind of OK that it works like that or am I missing something? E.g. I have to have an own branch for each of the environments?

Perhaps `ImageUpdateAutomation` is missing a `path` property?

**Comment by stefanprodan** [comment, marked as answer]
We are considering adding a `path` field to the `ImageUpdateAutomation.spec.update` so that users can set the same value as for `Kustomization.spec.path`.

**Reply by ilya-git** [reply]
That sounds great, in the meantime is it OK to run 2 parallel or will it break something? P.S. Do you have some timeframe for when it might be ready?

**Reply by stefanprodan** [reply]
The current workaround is to prefix/suffix your policies so they don't overlap.

**Reply by ilya-git** [reply]
Thanks :)

**Reply by stefanprodan** [reply]
Path implemented in https://github.com/fluxcd/image-automation-controller/pull/126

**Reply by ilya-git** [reply]
That was fast! Nice work :)
